packages/fxa-auth-server/lib/luaScripts/setRefreshToken_1.lua

```diff
@@ -0,0 +1,40 @@
+local uid = KEYS[1]
+local tokenId = ARGV[1]
+local token = ARGV[2]
+
+local function hgetall(key)
+  local kv = redis.call('hgetall', key)
+  local result = {}
+  local k
+  for i, v in ipairs(kv) do
+    if i % 2 == 1 then
+      k = v
+    else
+      result[k] = v
+    end
+  end
+  return result
+end
+
+local len = redis.call('hlen', uid)
+if len > 1 then
```

Comment:
So IIUC, if the user has a single refresh token, then we won't prune no matter how old it is. Is the intentional, and if so, why 1 as opposed to some other low number? (I don't thing there's anything wrong with this, just trying to understand the whole mental model here).

Reply:
Oh, you remind me that we should set / reset expiry on this whole key.

I wasn't really thinking very hard about the pruning criteria, just that we had some. We learned with access tokens not to rely on pruning with reads. I imagine we could make these numbers tunable with config.

Reply:
Makes sense 👍; it would also address those couple of comments in the code about relying on getters for lazy cleanup.